libpod/kube.go

```diff
@@ -107,8 +108,8 @@ func (p *Pod) GenerateForKube(ctx context.Context, getService bool) (*v1.Pod, []
 				pod.Spec.RestartPolicy = v1.RestartPolicyOnFailure
 			case define.RestartPolicyNo:
 				pod.Spec.RestartPolicy = v1.RestartPolicyNever
-			default: // some pod create from cmdline, such as "", so set it to Never
-				pod.Spec.RestartPolicy = v1.RestartPolicyNever
+			default: // some pod create from cmdline, such as "", so set it to "" as k8s automatically defaults to always
+				pod.Spec.RestartPolicy = ""
```

Comment:
That's potentially a breaking change.  Can you elaborate why that's needed for deployments?

Reply:
Deployments are only allowed to have a restart policy of `Always`. Setting the restart policy to `Never` by default in the generate kube yaml when the user doesn't specify a restart policy is also wrong, we shouldn't be setting a restart policy and let k8s handle it by using its own defaults. This is a bug imo as it breaks k8s convention.